components/camel-platform-http-main/src/main/java/org/apache/camel/component/platform/http/main/DefaultMainHttpServerFactory.java

```diff
@@ -59,6 +66,68 @@ public Service newHttpServer(HttpServerConfigurationProperties configuration) {
         server.setUploadEnabled(configuration.isUploadEnabled());
         server.setUploadSourceDir(configuration.getUploadSourceDir());
 
+        if (configuration.authentication().isEnabled()) {
+            configureAuthentication(server, configuration);
+        }
+
         return server;
     }
+
+    private void configureAuthentication(MainHttpServer server, HttpServerConfigurationProperties configuration) {
+        AuthenticationConfig authenticationConfig = server.getConfiguration().getAuthenticationConfig();
+
+        HttpServerAuthenticationConfigurationProperties authenticationProperties = configuration.getAuthentication();
+        Optional<MainAuthenticationConfigurer> authenticationConfigurer
+                = findAuthenticationConfigurerByConfigurationProperties(authenticationProperties);
+
+        authenticationConfigurer.ifPresentOrElse(
+                (configurer -> configurer.configureAuthentication(authenticationConfig, authenticationProperties)),
+                (() -> {
+                    throw new RuntimeException(
+                            "Authentication for camel-platform-http-main is enabled but no complete authentication configuration is found.");
+                }));
+    }
+
+    private Optional<MainAuthenticationConfigurer> findAuthenticationConfigurerByConfigurationProperties(
+            HttpServerAuthenticationConfigurationProperties authenticationConfigurationProperties) {
+
+        MainAuthenticationConfigurer result = null;
+        for (String authenticationTypeName : HttpServerAuthenticationConfigurationProperties.SUPPORTED_AUTHENTICATION_TYPES) {
+            if (authenticationTypeIsEnabled(authenticationConfigurationProperties, authenticationTypeName)) {
+                try {
+                    if (result != null) {
+                        throw new RuntimeException(
+                                "Cannot configure authentication for MainHttpServer as more than one authentication configuration is present");
+                    }
+                    String configurerQualifiedName = MainAuthenticationConfigurer.class.getPackageName() + "."
+                                                     + authenticationTypeName + "AuthenticationConfigurer";
+                    result = (MainAuthenticationConfigurer) Class
+                            .forName(configurerQualifiedName)
+                            .getDeclaredConstructor()
+                            .newInstance();
+                } catch (Exception e) {
+                    throw new RuntimeException(
+                            "Could not create MainAuthenticationConfigurer for authentication type " + authenticationTypeName,
+                            e);
+                }
+            }
+        }
+        return Optional.ofNullable(result);
+    }
+
+    private boolean authenticationTypeIsEnabled(
```

Comment:
Camel has some ReflectionHelper code or similar that can help with this kind of code.